specification/storage/data-plane/Microsoft.StorageDataLake/stable/2019-10-31/DataLakeStorage.json

```diff
@@ -0,0 +1,1922 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "description": "Azure Data Lake Storage provides storage for Hadoop and other big data workloads.",
+    "title": "Azure Data Lake Storage REST API",
+    "version": "2019-10-31",
+    "x-ms-code-generation-settings": {
+      "internalConstructors": true,
+      "name": "DataLakeStorageClient"
+    }
+  },
+  "x-ms-parameterized-host": {
+    "hostTemplate": "{accountName}.{dnsSuffix}",
+    "parameters": [
+      {
+        "$ref": "#/parameters/accountName"
+      },
+      {
+        "$ref": "#/parameters/dnsSuffix"
+      }
+    ]
+  },
+  "schemes": [
+    "http",
+    "https"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "tags": [
+    {
+      "name": "Account Operations"
+    },
+    {
+      "name": "Filesystem Operations"
+    },
+    {
+      "name": "File and Directory Operations"
+    }
+  ],
+  "parameters": {
+    "Version": {
+      "description": "Specifies the version of the REST protocol used for processing the request. This is required when using shared key authorization.",
+      "in": "header",
+      "name": "x-ms-version",
+      "required": false,
+      "type": "string",
+      "x-ms-parameter-location": "client"
+    },
+    "accountName": {
+      "description": "The Azure Storage account name.",
+      "in": "path",
+      "name": "accountName",
+      "required": true,
+      "type": "string",
+      "x-ms-skip-url-encoding": true,
+      "x-ms-parameter-location": "client"
+    },
+    "dnsSuffix": {
+      "default": "dfs.core.windows.net",
+      "description": "The DNS suffix for the Azure Data Lake Storage endpoint.",
+      "in": "path",
+      "name": "dnsSuffix",
+      "required": true,
+      "type": "string",
+      "x-ms-skip-url-encoding": true,
+      "x-ms-parameter-location": "client"
+    }
+  },
+  "definitions": {
+    "DataLakeStorageError": {
+      "properties": {
+        "error": {
+          "description": "The service error response object.",
+          "properties": {
+            "code": {
+              "description": "The service error code.",
+              "type": "string"
+            },
+            "message": {
+              "description": "The service error message.",
+              "type": "string"
+            }
+          }
+        }
+      }
+    },
+    "Path": {
+      "properties": {
+        "name": {
+          "type": "string"
+        },
+        "isDirectory": {
+          "default": false,
+          "type": "boolean"
+        },
+        "lastModified": {
+          "type": "string"
+        },
+        "eTag": {
+          "type": "string"
+        },
+        "contentLength": {
+          "type": "integer",
+          "format": "int64"
+        },
+        "owner": {
+          "type": "string"
+        },
+        "group": {
+          "type": "string"
+        },
+        "permissions": {
```

Comment:
The plural in the property name usually implies that the value is a collection (list etc.).